chapter2.adoc

```diff
@@ -1,214 +1,207 @@
 
-In addition, the `mprv` and `mpp` fields take effect exclusively when the *debug access privilege* is in machine mode.
+The subsequent subsections describe how external debug is authorized by <<mdbgctl, M-mode debug control>> and <<submdbgctl, supervisor domain debug control>>.
 
-[[maxdbgpriv]]
-[options="header"]
-.Determining maximum debug access privilege with mdbgen[i] and sdedbgalw
-|=========================================
-| mdbgen[i] | sdedbgalw | Maximum debug privilege 
-| 1      | x      | M                 
-| 0      | 1      | S(HS)             
-| 0      | 0      | n/a               
-|=========================================
+[NOTE]
+A pending request to enter Debug Mode can dynamically change from a disallowed state to an allowed state due to updates in debug controls. For example, once the software completes executing confidential code, it can grant debuggability for an external debugger. Afterwards, the software can enter a while(1) loop, waiting for the debugger to take control and break out of the loop.
 
 [[mdbgctl]]
-==== Machine Mode Debug Control
+==== M-mode Debug Control
 
-An input port, named mdbgen[i], is introduced to control the debuggability of machine mode for each hart i. This signal is transmitted to the hart and its corresponding debug access control logic. When mdbgen[i] is set to 1, the debug operations outlined in <<dbops>> are permitted when hart i executes in machine mode. Moreover the following rules apply:
+An input port, named mdbgen[i], is introduced to control the debuggability of M-mode for each hart i. This signal is propagated to the hart i and Debug Module as depicted in <<extdbg>>. When mdbgen[i] is set to 1, the following rules apply:
 
-- The <<dbgaccpriv, debug access privilege>> for the hart can be configured to any privilege level 
-- If register access without halting the hart is supported, this access carries the privilege of machine mode.
+- The <<dbgaccpriv, debug access privilege>> for the hart can be configured to any legal privilege level
+- The <<dbops, debug operations>> are permitted when hart i executes in all modes
+- Register access without halting the hart carries M-mode privilege if supported
 
-When mdbgen[i] is set to 0, debug operations in machine mode are prohibited for hart i. Any attempt to halt the hart and bring it into Debug Mode will remain pending, and triggers configured to enter Debug Mode will neither fire nor match in machine mode.
+When mdbgen[i] is set to 0, the <<dbops, debug operations>> are disallowed and the <<dbgdisallowed, behaviros>> applies when the hart runs in M-mode. 
 
 [NOTE]
-For the homogeneous computing system, the implementation can consolidate all mdbgen[i] into a single port to enforce unified debug policy across all harts.
+For a homogeneous computing system, the implementation can consolidate all mdbgen[i] into a single port to enforce a unified debug policy across all harts.
```

Comment:
Thanks for the suggestion. The behavior of mdbgen is already covered in previous paragraph, I will extend the note to be more clear:

>Since each hart has an input port for mdbgen, the implementation can choose to group several harts together and use one signal to drive their ports or assign each hart a dedicated signal.

What do you think of it?

Reply:
LGTM